File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach2/ui/MainViewModel.java
package alireza.nezami.dependency_injection_without_dagger.java.approach2.ui;


import alireza.nezami.dependency_injection_without_dagger.java.approach2.data.local.DatabaseService;
import alireza.nezami.dependency_injection_without_dagger.java.approach2.data.remote.NetworkService;

public class MainViewModel {

    private DatabaseService databaseService;
    private NetworkService networkService;

    public MainViewModel(DatabaseService databaseService, NetworkService networkService) {
        this.databaseService = databaseService;
        this.networkService = networkService;
    }

    public String getSomeData() {
        return databaseService.getDummyData() + " : " + networkService.getDummyData();
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach2/ui/MainViewModel.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach2.ui

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import alireza.nezami.dependency_injection_without_dagger.R
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach2.data.local.DatabaseService
import alireza.nezami.dependency_injection_without_dagger.kotlin.approach2.data.remote.NetworkService
import android.widget.TextView
import android.app.Application

class MainViewModel(
    private val databaseService: DatabaseService?,
    private val networkService: NetworkService?
) {
    val someData: String
        get() = databaseService!!.dummyData + " : " + networkService!!.dummyData
}